### PR TITLE
[V3] Smaller default header - Tall mode

### PR DESCRIPTION
**Pull request by kaelig**
The long awaited smaller header is here!

- Many fixes, tidied up the sub navigation
- Tall mode `o-header--tall` to get back to the original header size (on desktop only) in the case where you have a promo in the top right corner of the header
- Top logo is wrapped in a link so that the whole top left corner is actionable
- Added quick start step-by-step instructions in the readme

## Mobile: Before / after

screencapture-build-origami-ft-com-files-o-header-2-5-15-demos-header-html
screencapture-localhost-8080-demos-local-header-html

## Desktop

screen shot 2015-02-18 at 22 39 14

## Tall mode, useful when there's a promo

screen shot 2015-02-18 at 22 38 51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,22 @@
 Responsive FT page header.
 
 Note: this module used to be called `o-ft-header` which is why there are references to that name in verions 1 and 2 of the module.
+
+## Quick Start
+
+1. Load the CSS and the JavaScript in the document
+2. Load additional modules such as `o-ft-icons` and `o-fonts` to display FT branded icons and text
```
Comment (AlbertoElias) [inline README.md]: should that be o-icons?

File: README.md [hunk 2]
```diff
@@ -3,6 +3,22 @@
...
+3. Copy the markup from [one of the demos](http://registry.origami.ft.com/components/o-header) in the registry, and paste it in a document
+4. Set a `font-family` and a `overflow-x` on the document root:
+
+```scss
+html {
+	font-family: BentonSans, sans-serif;
+	overflow-x: hidden; // Prevent navigation menus from creating extra space on sides of the page
```
Comment (AlbertoElias) [inline README.md]: Is there no way round this?

File: README.md [hunk 3]
```diff
@@ -176,8 +190,16 @@ Add `<div class="o-header__inner">` in `<div class="o-header__container">`:
 </header>
 ```
 
-Note that o-header v3 relies on the o-grid v3 which introduces breaking changes.
+#### `.o-header__back-to-ft` icon
+
+```html
+- <i class="o-header__back-to-ft">Back to FT.com</i>
```
Comment (AlbertoElias) [inline README.md]: The back to ft icon isn't necessary if we add that icon to o-icons, isn't it?

File: demos/src/scss/demo.scss [hunk 4]
```diff
@@ -60,32 +61,34 @@ body {
...
 	@include oFtIconsBaseIconStyles();
 	display: block;
 	font-size: 20px;
-	line-height: 30px;
+	line-height: 24px;
 	color: #ffffff;
 }
 
-.demo__icon__search {
+.demo__icon--search {
```
Comment (AlbertoElias) [inline demos/src/scss/demo.scss]: If you're including o-icons via the build service in the demos, why not use them like that directly on the demo pages?

File: src/scss/nav/theme-primary.scss [hunk 5]
```diff
@@ -1,39 +1,42 @@
...
 	// scss-lint:disable SelectorDepth
-	[data-o-hierarchical-nav-level='1'] > li {
-		// Styling for the a tag in a selected state
-		&[aria-selected="true"] > a,
-		&[aria-controls][aria-expanded="true"] > a {
-			color: oColorsGetColorFor(o-header-item, text);
-			opacity: 0.7;
-		}
+	[data-o-hierarchical-nav-level='1'] {
+		> li {
```
Comment (AlbertoElias) [inline src/scss/nav/theme-primary.scss]: Why is the extra selector needed?

File: src/scss/nav/theme-tools.scss [hunk 6]
```diff
@@ -27,8 +30,8 @@
 
 		> li {
 			// Center elements vertically
-			height: 80px;
 			vertical-align: middle;
+			position: static; // Prevents sub menu from positionning themselves relative to the <li>
```
Comment (AlbertoElias) [inline src/scss/nav/theme-tools.scss]: Why is that important?

**Comment by AlbertoElias** [comment]
Left a few comments. Rest is cool!

File: src/scss/nav/theme-primary.scss (hunk 5 above)
Comment (kaelig) [inline src/scss/nav/theme-primary.scss]: Good catch, not sure why I did that!

**Pull request merged**